incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiServerCompositeBindingAdapter.java

```diff
@@ -0,0 +1,493 @@
+        }
+        return request;
+    }
+
+    private <C> BindingConfigBuilder<C> injectHttpServerRoutes(
+        BindingConfigBuilder<C> binding,
+        OpenApi openApi,
+        String guardName,
+        Map<String, String> securitySchemes)
+    {
+        for (String item : openApi.paths.keySet())
+        {
+            PathView path = PathView.of(openApi.paths.get(item));
+            for (String method : path.methods().keySet())
+            {
+                binding
+                    .route()
+                        .exit("http_client0")
+                        .when(HttpConditionConfig::builder)
+                            .header(":path", item.replaceAll("\\{[^}]+\\}", "*"))
+                            .header(":method", method)
+                            .build()
+                        .inject(route -> injectHttpServerRouteGuarded(route, path, method, guardName, securitySchemes))
+                        .build();
+            }
+        }
+        return binding;
+    }
+
+    private <C> RouteConfigBuilder<C> injectHttpServerRouteGuarded(
+        RouteConfigBuilder<C> route,
+        PathView path,
+        String method,
+        String guardName,
+        Map<String, String> securitySchemes)
+    {
+        final List<Map<String, List<String>>> security = path.methods().get(method).security;
+        final boolean hasJwt = securitySchemes.isEmpty();
+
+        if (security != null)
+        {
+            for (Map<String, List<String>> securityItem : security)
+            {
+                for (String securityItemLabel : securityItem.keySet())
+                {
+                    if (hasJwt && "jwt".equals(securitySchemes.get(securityItemLabel)))
+                    {
+                        route
+                            .guarded()
+                                .name(guardName)
+                                .inject(guarded -> injectGuardedRoles(guarded, securityItem.get(securityItemLabel)))
+                                .build();
+                    }
+                }
+            }
+        }
+        return route;
+    }
+
+    private <C> GuardedConfigBuilder<C> injectGuardedRoles(
+        GuardedConfigBuilder<C> guarded,
+        List<String> roles)
+    {
+        for (String role : roles)
+        {
+            guarded.role(role);
+        }
+        return guarded;
+    }
+
+    private SchemaView resolveSchemaForJsonContentType(
+        Map<String, MediaType> content,
+        OpenApi openApi)
+    {
+        MediaType mediaType = null;
+        if (content != null)
+        {
+            for (String contentType : content.keySet())
+            {
+                if (jsonContentType.reset(contentType).matches())
+                {
+                    mediaType = content.get(contentType);
+                    break;
+                }
+            }
+        }
+        return mediaType == null ? null : SchemaView.of(openApi.components.schemas, mediaType.schema);
+    }
```

Comment:
This helper method should go after all the injectXxx methods for improved readability.